src/Symfony/Component/Console/Tester/CommandTester.php

```diff
@@ -48,7 +48,7 @@ public function __construct(Command $command)
      *
      * @return int The command exit code
      */
-    public function execute(array $input, array $options = array())
+    public function execute(array $input = array(), array $options = array()): int
```

Comment:
Changing the signature is a BC break, any subclass overriding this method would be broken, not something acceptable

Reply:
This is true, but according to @xabbuh the Symfony test suite does not apply BC promise

https://github.com/symfony/symfony/issues/25358#issuecomment-349721982

Since this is in the `Tester` namespace, I assume this is also part of the symfony test suite.

Reply:
The BC promise does apply for this class as its intended to be used for testing console applications. From http://symfony.com/doc/current/contributing/code/bc.html#using-our-classes:

> Classes, properties and methods that bear the tag @internal as well as the classes located in the various *\\Tests\\ namespaces are an exception to this rule. They are meant for internal use only and should not be accessed by your own code.

Reply:
Just to explain my way of thinking: I thought because of `Tester` it would fall under `test suite`.

But after your explanation, I've got to agree with you. This would indeed be a BC. I could change the PR here to state that it's a BC break, but I think it will take some time until Symfony5 comes out, so... dunno, close this PR? State that this PR breaks BC? Wait for sth. official?

Reply:
No intentional BC break, and always trigger a deprecation first.
But in this case, I don't see the need for it... So I'd suggest to close :)